cmd/contrib/scripts/config_dir_getter.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/mitchellh/go-homedir"
+)
+
+// Retrieve the config directory to read or write to
+func main() {
+	opConfigDir, _ := os.LookupEnv("OP_CONFIG_DIR")
+	xdgConfigHome, _ := os.LookupEnv("XDG_CONFIG_HOME")
+	home, _ := homedir.Dir()
+
+	// This logic is based on the order of precedence outlined in the CLI documentation:
+	// https://developer.1password.com/docs/cli/config-directories
+	configDirPaths := []string{}
+	if opConfigDir != "" {
+		configDirPaths = append(configDirPaths, opConfigDir)
+	}
+	if home != "" {
+		// Legacy home
+		configDirPaths = append(configDirPaths, filepath.Join(home, ".op"))
+	}
+	if xdgConfigHome != "" {
+		// Legacy xdg
+		configDirPaths = append(configDirPaths, filepath.Join(xdgConfigHome, ".op"))
+	}
+	if home != "" {
+		// New home
+		configDirPaths = append(configDirPaths, filepath.Join(home, ".config", "op"))
+	}
+	if xdgConfigHome != "" {
+		// New xdg
+		configDirPaths = append(configDirPaths, filepath.Join(xdgConfigHome, "op"))
+	}
+
+	for _, configDir := range configDirPaths {
+		fileInfo, err := os.Stat(configDir)
+		if err == nil && fileInfo.IsDir() {
+			fmt.Print(configDir)
+			return
+		}
+	}
+
+	// If we reach this point then none of those directories exist (op is executed
+	// for the first time). Default to the last entry in the list.
```

Comment:
I suggest we should skip this step, if an environment variable has been set, but the directory has not been found.

Reply:
Would you mind elaborating? 

From my understanding, this code is necessary. If a user has just started working with the 1Password CLI for example and runs `make <plugin>/build` for the first time, in which case no directories would be found, this block of code should run and create the directory.

Reply:
In its current form, the code would still create one of our predefined locations, if the directory under `OP_CONFIG_DIR` doesn't exist. That is, if the user mistypes the value under `OP_CONFIG_DIR`, they would expect no side effects if the directory pointed to by the envvar doesn't exist. However, a directory (the last in the list of viable dirs) gets created, if no other exists.

Reply:
Sorry I'm still a bit confused about the desired behaviour. If the user has defined `OP_CONFIG_DIR` but the directory doesn't exist, would you want the `make <plugin>/build` command to fail entirely?

If the end of the `config_dir_getter.go` script is reached and nothing is returned/printed, then when `make <plugin>/build` is called, in the Makefile `config_dir` evaluates to a blank string and `plugins_dir` evaluates to `/plugins/local`. A command is then called to create a directory at `/plugins/local`, which is even more unexpected than creating a new directory at one of the predefined locations, `~/.op/plugins/local` for instance. 

Since error handling isn't really something you can do within a Makefile, the current implementation was the best solution I could think of. Also, I should note that I matched the logic currently used in the `op` repo for consistency.

Reply:
I re-tested how `op` searches for a config directory and it seems that if `OP_CONFIG_DIR` is set, the location specified by `OP_CONFIG_DIR` is used even if other predefined config directories already exist. This was something I missed when I first created this PR - I pushed a change to match the behaviour seen in `op` since the behaviour in both repos should match.